Doc/library/socket.rst

```diff
@@ -595,6 +595,65 @@ The following functions all create :ref:`socket objects <socket-objects>`.
    .. versionchanged:: 3.2
       *source_address* was added.
 
+.. function:: bind_socket(address, *, family=AF_UNSPEC, type=SOCK_STREAM, backlog=128, reuse_addr=None, reuse_port=False, flags=None, hybrid_ipv46=False)
+
+   Convenience function which aims at automating all the typical steps needed
+   when creating a server socket.
+   It creates a socket bound to *address* (a 2-tuple ``(host, port)``) and
```

Comment:
No, also for SOCK_DGRAM (and this function supports only these 2).

Reply:
@njsmith said something about SOCK_UNIX. Does it make any sense to support SOCK_UNIX as well?

Reply:
I think that would be better served as a separate function. Mostly because: 
1) the address would no longer be a `(host, ip)` tuple but a path
2) a UNIX socket may require it's own parameters (e.g. `mode`). It is true that we already have a parameter which applies to `SOCK_STREAM` sockets only (the `backlog`), but `SOCK_STREAM` are much more used than `AF_UNIX`.